src/core/toga/widgets/canvas.py

```diff
@@ -1,20 +1,297 @@
+from contextlib import contextmanager
 from .base import Widget
+from math import pi
 
 
 class Canvas(Widget):
-    """"""
+    """Create new canvas
+
+    Args:
+        id (str):  An identifier for this widget.
+        style (:class:`colosseum.CSSNode`): An optional style object. If no
+            style is provided then a new one will be created for the widget.
+        factory (:obj:`module`): A python module that is capable to return a
+            implementation of this class with the same name. (optional &
+            normally not needed)
+    """
 
     def __init__(self, id=None, style=None, factory=None):
+        super().__init__(id=id, style=style, factory=factory)
+
+        # Create a platform specific implementation of Canvas
+        self._impl = self.factory.Canvas(interface=self)
+
+        self.rehint()
+
+    @contextmanager
+    def save_restore(self):
```

Comment:
I'm not wild about the naming here. It definitely describes what will be done, but not why. `context()` might be a better name - it would allow for the following usage:
```
with canvas.context() as context:
     context.line(...)
```
Thoughts?